src/components/listing-components/login-dialog.tsx

```diff
@@ -80,86 +87,211 @@ const loginDialog: React.FC<LoginDialogProps> = ({ open, onClose, onLogin }) =>
   const validateLogin = (usernameS: string, passwordS: string) => {
     return usernameS.trim() !== "" && passwordS.trim() !== "";
   };
-  
+
   /**
    * Handle login
+   * @param event 
    */
-  const handleLogin = () => {
+  const handleLogin = async (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
     const isValidLogin = validateLogin(username, password);
-  
     if (isValidLogin) {
-      onLogin();
-      onClose();
+      const { accessToken: newAccessToken, refreshToken: newRefreshToken, error } = await fetchLogin({
+        selectedSite: siteList.find(siteItem => siteItem.id === site),
+        username: username,
+        password: password
+      });
+
+      if (error) {
+        setLoginError(error);
+      } else {
+        setAccessToken(newAccessToken);
+        setRefreshToken(newRefreshToken);
+        onLogin();
+        onClose(loginError, undefined);
+      }
     } else {
       setLoginError(strings.errorHandling.listingScreenLogin.loginFailed);
     }
   };
 
   /**
+   * Token refresh logic
+   */
+  const tokenRefresh = async () => {
+    const { accessToken: newAccessToken, refreshToken: newRefreshToken, error } = await handleTokenRefresh({
+      selectedSite: siteList.find(siteItem => siteItem.id === site),
+      refreshToken: refreshToken
+    });
+    if (error) {
+      errorContext.setError(error);
+    } else {
+      setAccessToken(newAccessToken);
+      setRefreshToken(newRefreshToken);
+    }
+  };
+  
+  // Var for token refresh timer
+  let expirationTime = 0;
```

Comment:
You should not use mutable parameters directly inside react component structure, as the value cannot really be updated after a single render of the function component, and it is kinda anti-pattern anyways, as React development relies on immutable state of the components.